src/oneD/IonFlow.cpp

```diff
@@ -0,0 +1,416 @@
+//! @file IonFlow.cpp
+
+// This file is part of Cantera. See License.txt in the top-level directory or
+// at http://www.cantera.org/license.txt for license and copyright information.
+
+#include "cantera/oneD/IonFlow.h"
+#include "cantera/oneD/StFlow.h"
+#include "cantera/base/ctml.h"
+#include "cantera/transport/TransportBase.h"
+#include "cantera/numerics/funcs.h"
+
+using namespace std;
+
+namespace Cantera
+{
+
+IonFlow::IonFlow(IdealGasPhase* ph, size_t nsp, size_t points) :
+    FreeFlame(ph, nsp, points),
+    m_phase(1),
+    m_inletVoltage(0),
+    m_outletVoltage(0),
+    m_kElectron(Undef)
+{
+    // make a local copy of species charge
+    for (size_t k = 0; k < m_nsp; k++) {
+        m_speciesCharge.push_back(m_thermo->charge(k));
+    }
+
+    // Find indices for charge of species
+    for (size_t k = 0; k < m_nsp; k++){
+        if (m_speciesCharge[k] != 0){
+            m_kCharge.push_back(k);
+        } else {
+            m_kNeutral.push_back(k);
+        }
+    }
+
+    // Find the index of electron 
+    if (m_thermo->speciesIndex("E") != npos ) {
+        m_kElectron = m_thermo->speciesIndex("E");
+    }
+
+    // mass fraction bounds (strict bound for ions)
+    for (size_t k : m_kCharge) {
+        setBounds(c_offset_Y+k, -1.0e-20, 1e-5);
```

Comment:
It seems 1.0e5 is okay. But I found that -1.0e-20 is needed for better convergence.